src/components/HistoryTable/utils.ts

```diff
@@ -31,24 +40,38 @@ export const processCommits = (
           type: rowType.COMMIT,
           commit: version,
           date: version.createTime,
+          selected: version.order === selectedCommitOrder,
           rowHeight: COMMIT_HEIGHT,
         });
+
+        if (version.order === selectedCommitOrder) {
+          selectedCommitRowIndex = processedCommits.length - 1;
+        }
       } else {
         processedCommits.push({
           type: rowType.DATE_SEPARATOR,
           date: version.createTime,
           rowHeight: DATE_SEPARATOR_HEIGHT,
+          selected: false,
         });
         processedCommits.push({
           type: rowType.COMMIT,
           commit: version,
           date: version.createTime,
+          selected: version.order === selectedCommitOrder,
           rowHeight: COMMIT_HEIGHT,
         });
+        if (version.order === selectedCommitOrder) {
+          selectedCommitRowIndex = processedCommits.length - 1;
+        }
```

Comment:
If I had to guess I think the `selectedCommitRowIndex` is off by one? This only happens from the `/commits` page, not on any Task History buttons

https://user-images.githubusercontent.com/47064971/160643333-fdc26631-4d94-492b-b802-7a8499a9a2a4.mov